utils.py

```diff
@@ -0,0 +1,324 @@
+import xlmhg
+import csv
+import scipy.io
+import pandas as pd
+import numpy as np
+import os
+import multiprocessing as mp
+import glob
+import matplotlib.pyplot as plt
+
+
+def check_data_type(data_path, dataset_name):
+    """
+    checks if single cell or spatial data
```

Comment:
Method docstring should be:
Summary line (not exceeding 80 char).
Body
Args
Returns

Example:
```
def check_data_type(data_path: str, dataset_name: str) -> str:
"""Checks if single cell or spatial data.

Some other relevant things you want to add in this context.

Args:
  dataset_name: dataset folder name.
  data_path: path for dataset folder.

Returns:
  the data type of the provided dataset, "scRNAseq" or "spatial".
"""
```